services/agent/src/simcore_service_agent/modules/task_monitor.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+import logging
+import traceback
+from collections import deque
+from contextlib import suppress
+from dataclasses import dataclass
+from typing import Any, Awaitable, Callable, Coroutine, Final, Optional
+
+from fastapi import FastAPI
+from pydantic import PositiveFloat, PositiveInt
+
+from ..core.settings import ApplicationSettings
+from .volumes_cleanup import backup_and_remove_volumes
+from servicelib.logging_utils import log_context
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TASK_WAIT_ON_ERROR: Final[PositiveInt] = 10
+
+
+@dataclass
+class _TaskData:
+    callable: Callable
+    args: Any
+    repeat_interval_s: Optional[PositiveFloat]
+
+
+async def _task_runner(task_data: _TaskData) -> None:
+    with log_context(logger, logging.INFO, msg=f"'{task_data.callable.__name__}'"):
+        while True:
+            coroutine: Coroutine = task_data.callable(*task_data.args)
+            try:
+                await coroutine
+            except Exception as e:  # pylint: disable=broad-except
+                logger.error(
+                    "Had an error while running '%s':\n%s",
+                    coroutine.__name__,
+                    "\n".join(traceback.format_tb(e.__traceback__)),
```

Comment:
``logger.exception`` includes traces of the caught ``e``  ... no need to add this traceback. There are also extra options make it shorter traceback etc. PLease read the loggers manual